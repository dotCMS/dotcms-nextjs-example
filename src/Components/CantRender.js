import React from 'react';
import { Alert } from 'reactstrap';

const CantRender = (props) => {
    return (
        <Alert className="alert" color={props.color || 'danger'}>
            <h4 className="alert-heading">{props.title}</h4>
            {props.children}
        </Alert>
    );
};

export default CantRender;