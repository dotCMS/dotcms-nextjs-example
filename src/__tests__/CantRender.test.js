import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import CantRender from '../Components/CantRender';

describe('<CantRender />', () => {
    it('should render default values correctly', () => {
        const wrapper = shallow(
            <CantRender title="Hello World">
                <p>Hola Mundo</p>
            </CantRender>
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('should render color values correctly', () => {
        const wrapper = shallow(
            <CantRender color="warning" title="Hello World">
                <p>Hola Mundo</p>
            </CantRender>
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
});
