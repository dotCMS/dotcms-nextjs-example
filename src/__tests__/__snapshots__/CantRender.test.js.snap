// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CantRender /> should render color values correctly 1`] = `
<Alert
  className="alert"
  closeAriaLabel="Close"
  color="warning"
  fade={true}
  isOpen={true}
  tag="div"
  transition={
    Object {
      "appear": true,
      "baseClass": "fade",
      "baseClassActive": "show",
      "enter": true,
      "exit": true,
      "in": true,
      "mountOnEnter": false,
      "onEnter": [Function],
      "onEntered": [Function],
      "onEntering": [Function],
      "onExit": [Function],
      "onExited": [Function],
      "onExiting": [Function],
      "tag": "div",
      "timeout": 150,
      "unmountOnExit": true,
    }
  }
>
  <h4
    className="alert-heading"
  >
    Hello World
  </h4>
  <p>
    Hola Mundo
  </p>
</Alert>
`;

exports[`<CantRender /> should render default values correctly 1`] = `
<Alert
  className="alert"
  closeAriaLabel="Close"
  color="danger"
  fade={true}
  isOpen={true}
  tag="div"
  transition={
    Object {
      "appear": true,
      "baseClass": "fade",
      "baseClassActive": "show",
      "enter": true,
      "exit": true,
      "in": true,
      "mountOnEnter": false,
      "onEnter": [Function],
      "onEntered": [Function],
      "onEntering": [Function],
      "onExit": [Function],
      "onExited": [Function],
      "onExiting": [Function],
      "tag": "div",
      "timeout": 150,
      "unmountOnExit": true,
    }
  }
>
  <h4
    className="alert-heading"
  >
    Hello World
  </h4>
  <p>
    Hola Mundo
  </p>
</Alert>
`;
